lib/libc/misc/lib_sendfile.c

```diff
@@ -216,6 +216,17 @@ ssize_t sendfile(int outfd, int infd, off_t *offset, size_t count)
 			 */
 
 			wrbuffer = iobuffer;
+
+			/* Check the limitation of ntransferred (SSIZE_MAX) */
+
+			if ((size_t)ntransferred + nbytesread > SSIZE_MAX) {
+
+				/* Re-calculate the size which will be sent not to overflow maximum value of ssize_t */
+
+				nbytesread = SSIZE_MAX - ntransferred;
+				endxfr = true;
```

Comment:
In my opinion, This should not be set here.
```
if ((size_t)ntransferred + nbytesread > SSIZE_MAX) {
    nbytesread = SSIZE_MAX - ntransferred;
}
do {
    ...
    nbyteswritten = write(outfd, wrbuffer, nbytesread);
    if (ntransffered == SSIZE_MAX) {
          break;
    }
}
```

Reply:
`break` is not enough. It will escape the do loop but it will continue the for loop even it sent the maximum size already.

Reply:
And before meet `if (ntransffered == SSIZE_MAX)` statement, `nbytesread` becomes zero and do loop will be end by while conditional.
I mean 
1. The sendfile function already sent the maximum size so that it should be end. But `break` can't stop the loop.
2. `break` statement is duplicated with while conditional and it is unreachable statement.

Reply:
I didn't see do-while sorry about it =)
But still it is not correct here, it should restrict size of read first.

Reply:
Fixed as you said.